homeassistant/components/barry/__init__.py

```diff
@@ -0,0 +1,49 @@
+"""The Barry App integration."""
+import logging
+
+from pybarry import Barry
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+from .const import DOMAIN
+
+PLATFORMS = [
+    "sensor",
+]
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up the Barry component."""
+
+    if DOMAIN not in config:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=config[DOMAIN],
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up a config entry."""
+
+    barry_connection = Barry(
+        access_token=entry.data[CONF_ACCESS_TOKEN],
+    )
+    hass.data[DOMAIN] = barry_connection
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
```

Comment:
```suggestion
    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
```

Reply:
I tried the suggested code but I am getting error as "Error while setting up barry platform for sensor". In contrast, when I keep code with lines 44 to 47, it is working file without any error.

Reply:
Its a helper method that does exactly the same... 🤷 
It just for reducing code.